src/GraphQL.MicrosoftDI/ScopedFieldResolver.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using GraphQL.Resolvers;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.MicrosoftDI
+{
+    /// <summary>
+    /// When resolving a field, this implementation calls
+    /// <see cref="IResolveFieldContext.RequestServices"/>.<see cref="ServiceProviderServiceExtensions.CreateScope(IServiceProvider)">CreateScope</see>
+    /// to create a dependency injection scope. Then it calls a predefined <see cref="Func{T, TResult}"/>, passing the scoped service provider
+    /// within <see cref="IResolveFieldContext.RequestServices"/>, and returns the result.
+    /// </summary>
+    public class ScopedFieldResolver<TReturnType> : FuncFieldResolver<TReturnType>
+    {
+        /// <summary>
+        /// Initializes a new instance that creates a service scope and runs the specified delegate when resolving a field.
+        /// </summary>
+        public ScopedFieldResolver(Func<IResolveFieldContext, TReturnType> resolver) : base(GetScopedResolver(resolver)) { }
+
+        private static Func<IResolveFieldContext, TReturnType> GetScopedResolver(Func<IResolveFieldContext, TReturnType> resolver)
+        {
+            return context =>
+            {
+                using (var scope = (context.RequestServices ?? throw new MissingRequestServicesException()).CreateScope())
```

Comment:
Where does `CreateScope` method come from?

Reply:
It comes from the abstractions import. But it’s only one or two lines of code - we could write it ourselves - except we still need a reference to IServiceScopeFactory

Reply:
It’s basically GetService<IServiceScopeFactory>().CreateScope()

Reply:
Found.